### ISSUE TITLE
S3 fails on Buckets with permissions scoped to prefixes

### ISSUE THREAD

**Issue by advanderveer** [restic/restic#1477]
Thank you for this great piece of software! I've run into an issue on S3 Buckets with tightly controlled permissions.

## Output of `restic version`
```
restic 0.8.0
compiled with go1.9 on darwin/amd64
```

## How did you run restic exactly?
env (I cleared all other variables):
```
AWS_ACCESS_KEY_ID=<key> 
AWS_SECRET_ACCESS_KEY=<secret>
AWS_SESSION_TOKEN=<token>
```

```
restic init --password-file=<my-password-file> -r=s3:s3.amazonaws.com/<my-bucket>/<my-prefix>
```

```
create backend at s3:s3.amazonaws.com/<my-bucket>/chunks/<my-user>/hhh failed: client.BucketExists: Access Denied
```

## What backend/server/service did you use to store the repository?
hosted s3

## Expected behavior
I created the bucket by hand and tightly controlled the permissions for clients. The s3:HeadBucket operations ([client.BucketExists](https://github.com/minio/minio-go/blob/3419703769e0de7e06fab876ce54605df621f017/api-stat.go)) will return permission denied in these cases even though all operations necessary for backup/restore are allowed. 

I would expect restic to allow to disable this behaviour through an option and trust I setup the bucket correctly myself.  

## Actual behavior
It checks for existence in order to try and [create a bucket](https://github.com/restic/restic/blob/1ebf0e8de8f406f11c3d35b8479810433a1c0b71/internal/backend/s3/s3.go#L143), both calls will fail in tightly controlled s3 environments.

## Steps to reproduce the behavior
A user with the following permissions would cause the issue:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:HeadBucket"
            ],
            "Resource": [
                "arn:aws:s3:::<buket-name>"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::<bucket-name>"
            ],
            "Condition": {
                "StringLike": {
                    "s3:prefix": [
                        "chunks/my-user"
                    ]
                }
            }
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:GetObject",
                "s3:PutObject",
                "s3:DeleteObject"
            ],
            "Resource": [
                "arn:aws:s3:::<bucket-name>/chunks/my-user",
                "arn:aws:s3:::<bucket-name>/chunks/my-user/*"
            ]
        }
    ]
}
```

## Do you have any idea what may have caused this?
On AWS s3 it is more common to share a bucket between users as there is an account limit on the number of buckets (100). In such cases it makes sense to limit the operations on a certain prefix to a certain user. In such cases s3:HeadBucket will return forbidden and creation fails.

## Do you have an idea how to solve the issue?
An option that configures the backend not to try and attempt any actions on the bucket level.

**Comment by fd0** [comment]
Thanks for the report, we can indeed handle the situation better.

**Comment by MichaelEischer** [comment]
Closing this as it seems to have been fixed in #1648 which is included in restic 0.9.0. Feel free to comment if that's not the case.
